src/codegen.cpp

```diff
@@ -8136,7 +8136,7 @@ extern "C" void jl_init_llvm(void)
         };
 
     jl_default_debug_info_kind = (int) DICompileUnit::DebugEmissionKind::FullDebug;
-    imaging_mode = jl_options.image_codegen || (jl_generating_output() && !jl_options.incremental);
+    imaging_mode = jl_options.image_codegen || (jl_generating_output());// && !jl_options.incremental);
```

Comment:
Why do you need to set this? Was this because otherwise we drop the cache?

Reply:
Yes, otherwise the inferred code frequently gets discarded, see https://github.com/JuliaLang/julia/blob/e3b681c6872d87d74db63dd7ec3da2beee8e3ae4/src/codegen.cpp#L7816. And since we can't (yet) cache the native code, that's a bad outcome.

Reply:
Can we instead check at that site https://github.com/JuliaLang/julia/blob/e3b681c6872d87d74db63dd7ec3da2beee8e3ae4/src/codegen.cpp#L7814 for jl_options.incremental instead of setting `imaging_mode` (it produces slightly worse code)

Reply:
Sure, that's a good idea.